sql/transactions/transactions.go

```diff
@@ -410,3 +411,35 @@ func TransactionInBlock(
 	}
 	return bid, rst, nil
 }
+
+func IterateTransactionsOps(
+	db sql.Executor,
+	operations builder.Operations,
+	fn func(tx *types.MeshTransaction, result *types.TransactionResult) bool,
+) error {
+	var derr error
+	_, err := db.Exec(`select distinct tx, header, layer, block, timestamp, id, result 
+		from transactions
+		left join transactions_results_addresses on id=tid
+		where result is not null`+builder.FilterFrom(operations),
+		builder.BindingsFrom(operations),
+		func(stmt *sql.Statement) bool {
+			var txId types.TransactionID
+			stmt.ColumnBytes(5, txId[:])
+			tx, derr := decodeTransaction(txId, stmt)
```

Comment:
`derr` is shadowed here - the outer `derr` variable will never be set since this line creates a local `derr`.

See for instance: https://go.dev/play/p/j8VU6SIeuSY

Reply:
Fixed